apl-utils/src/main/java/com/apollocurrency/aplwallet/apl/util/cache/CacheConfigurator.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c)  2018-2019. Apollo Foundation.
+ */
+
+package com.apollocurrency.aplwallet.apl.util.cache;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Preconditions;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+
+import java.util.Objects;
+import java.util.Optional;
+
+public class CacheConfigurator<K, V> implements CacheConfiguration<K, V> {
+    private String name;
+    private long elementSize;
+    private int cachePriority;
+    private int maxSize = -1;
+    private CacheBuilder cacheBuilder;
+    private Optional<CacheLoader<K, V>> optionalCacheLoader;
+
+    public CacheConfigurator(String name, long elementSize, int cachePriority) {
+        this(name, elementSize, cachePriority, null);
+    }
+
+    public CacheConfigurator(String name, long elementSize, int cachePriority, CacheLoader<K, V> cacheLoader) {
+        Objects.requireNonNull(name, "Cache name is NULL.");
+        this.name = name;
+        this.elementSize = elementSize;
+        this.cachePriority = cachePriority;
+        this.cacheBuilder = CacheBuilder.newBuilder();
+        this.optionalCacheLoader = Optional.ofNullable(cacheLoader);
```

Comment:
Looks like Optional here is not required. Just return Optional.ofNullable in getCacheLoader

Reply:
I'll return Optional in that method.